google-cloud-logging/src/test/java/com/google/cloud/logging/BaseSystemTest.java

```diff
@@ -74,19 +75,49 @@ protected static String createTimestampFilter(int hoursAgo) {
     return "timestamp>=\"" + RFC_3339.format(calendar.getTime()) + "\"";
   }
 
-  /** Helper to poll for logs until they are returned by the backend. */
-  protected static Iterator<LogEntry> waitForLogs(LogName logName) throws InterruptedException {
-    return waitForLogs(logName, 1);
+  protected static String appendResourceTypeFilter(
+      String currentFilter, MonitoredResource[] monitoredResources) {
+    StringBuilder filter = new StringBuilder(currentFilter);
+    if (monitoredResources != null && monitoredResources.length > 0) {
+      if (monitoredResources.length == 1) {
+        filter.append(" AND resource.type=");
+        filter.append(monitoredResources[0].getType());
+      } else {
+        filter.append(" AND resource.type=(");
+        filter.append(monitoredResources[0].getType());
+
+        // OR between all monitored resources we search
+        for (int i = 1; i < monitoredResources.length; i++) {
+          filter.append(" OR ");
+          filter.append(monitoredResources[i].getType());
+        }
+        filter.append(")");
+      }
+    }
+
+    return filter.toString();
   }
 
   /** Helper to poll for logs until they are returned by the backend. */
-  protected static Iterator<LogEntry> waitForLogs(LogName logName, int minLogs)
+  protected static Iterator<LogEntry> waitForLogs(
+      LogName logName, MonitoredResource[] monitoredResources, int minLogs)
       throws InterruptedException {
-    String filter = createEqualityFilter("logName", logName) + " AND " + createTimestampFilter(1);
-    Logging.EntryListOption[] options = {Logging.EntryListOption.filter(filter)};
+    StringBuilder filter = new StringBuilder();
+    filter.append(createTimestampFilter(1));
+    filter.append(" AND ");
+    filter.append(createEqualityFilter("logName", logName));
+
+    String monitoredResourceFilter =
+        appendResourceTypeFilter(filter.toString(), monitoredResources);
+
+    Logging.EntryListOption[] options = {Logging.EntryListOption.filter(monitoredResourceFilter)};
```

Comment:
Do you think we should change `Logging.EntryListOption.filter` to use a 1 day timestamp filter by default? That's what [I did in Python](https://github.com/googleapis/python-logging/issues/71). It seems like a bad developer experience to hang on this function using the default options

Reply:
That's what I did here too - even a bit more restrictive filter of 1 hour back instead of 24 hours back. If you look at line 106 - I call createTimestampFilter(1) to restrict logs range, then on line 110 I add resourceType filter to further restrict the filter on top of time range.

Reply:
It looks like you changed the test, but should we change the function itself too?